### PR TITLE
Adding ResourceScope to ResourceExhaustedFailure

### PR DESCRIPTION
**Pull request by prathyushpv**
**What changed?**
Adding a new field ResourceScope in ResourceExhaustedFailure.

This PR depends on https://github.com/temporalio/api/pull/382. 

**Why?**
We need to differentiate whether a request has been rate limited because of namespace limit or system limit.
This will help us better understand why requests are rate limited and which limits to adjust.

**How did you test it?**

**Potential risks**
None

### CODE REVIEW

File: serviceerror/resourceExhausted.go [hunk 1]
```diff
@@ -34,15 +34,17 @@ type (
...
 		Cause   enumspb.ResourceExhaustedCause
+		Scope   enumspb.ResourceScope
 		Message string
 		st      *status.Status
 	}
 )
 
 // NewResourceExhausted returns new ResourceExhausted error.
-func NewResourceExhausted(cause enumspb.ResourceExhaustedCause, message string) error {
+func NewResourceExhausted(cause enumspb.ResourceExhaustedCause, scope enumspb.ResourceScope, message string) error {
```
Comment (cretz) [inline serviceerror/resourceExhausted.go:44]: While we may think this is only used by us, this is public API at https://pkg.go.dev/go.temporal.io/api/serviceerror and we shouldn't break compatibility IMO. Can we either have a `NewResourceExhaustedWithOptions` or request that those building this with a scope create the struct directly?

Comment (prathyushpv) [reply serviceerror/resourceExhausted.go:44]: Okay! I thought only server and go-sdk use this function. Let me make that change.

Comment (prathyushpv) [reply serviceerror/resourceExhausted.go]: I have removed this argument from this function. It is better to create this struct directly.

Comment (cretz) [reply serviceerror/resourceExhausted.go]: Thanks! Will approve when api repo PR merged and regen occurs and is merged into this branch

Comment (prathyushpv) [reply serviceerror/resourceExhausted.go]: Thanks @cretz! I have merge api changes..

**Review by cretz — APPROVED** [review]

**Pull request merged**